yt/frontends/arepo/fields.py

```diff
@@ -35,7 +34,22 @@ def __init__(self, ds, field_list, slice_info=None):
             GFM_SFT_units = "dimensionless"
         else:
             GFM_SFT_units = "code_length/code_velocity"
-        self.known_particle_fields += (("GFM_StellarFormationTime", (GFM_SFT_units, ["stellar_age"], None)), )
+        new_particle_fields = (("GFM_StellarFormationTime", (GFM_SFT_units, ["stellar_age"], None)),
+                               ("MagneticField", ("code_magnetic", ["particle_magnetic_field"], None)),
+                               ("MagneticFieldDivergence",
+                                ("code_magnetic/code_length", ["magnetic_field_divergence"], None)),
+                               ("GFM_Metallicity", ("", ["metallicity"], None)),
+                               ("GFM_Metals_00", ("", ["H_fraction"], None)),
+                               ("GFM_Metals_01", ("", ["He_fraction"], None)),
+                               ("GFM_Metals_02", ("", ["C_fraction"], None)),
+                               ("GFM_Metals_03", ("", ["N_fraction"], None)),
+                               ("GFM_Metals_04", ("", ["O_fraction"], None)),
+                               ("GFM_Metals_05", ("", ["Ne_fraction"], None)),
+                               ("GFM_Metals_06", ("", ["Mg_fraction"], None)),
+                               ("GFM_Metals_07", ("", ["Si_fraction"], None)),
+                               ("GFM_Metals_08", ("", ["Fe_fraction"], None)),
+                               )
+        self.known_particle_fields += new_particle_fields
```

Comment:
Why not adding these known fields directly in the initialisation line for `know_particle_fields` ?

Reply:
One of these fields has information that is determined at load time, so rather than messing with `known_particle_fields` twice in here I've decided to do it only once in `__init__`.